apis/v1alpha3/vspheremachine_conversion.go

```diff
@@ -39,6 +39,8 @@ func (src *VSphereMachine) ConvertTo(dstRaw conversion.Hub) error {
 
 	dst.Spec.AdditionalDisksGiB = restored.Spec.AdditionalDisksGiB
 	dst.Spec.TagIDs = restored.Spec.TagIDs
+	dst.Spec.PowerOffMode = infrav1beta1.VirtualMachinePowerOpModeHard
+	dst.Spec.GuestSoftPowerOffTimeout = nil
```

Comment:
You should be restoring the spec. 
What this is doing is that if a v1alpha3 or v1alpha4 client modifies the resource, then you are hardcoding PowerOpModeHard regardless of what was originally selected.

Reply:
Just read the test case code carefully and I hope I did it right this time.

Reply:
Yup looks good, thx!